velox/common/memory/Memory.cpp

```diff
@@ -195,22 +193,32 @@ std::shared_ptr<MemoryPool> MemoryManager::addLeafPool(
 bool MemoryManager::growPool(MemoryPool* pool, uint64_t incrementBytes) {
   VELOX_CHECK_NOT_NULL(pool);
   VELOX_CHECK_NE(pool->capacity(), kMaxMemory);
-  return arbitrator_->growMemory(pool, getAlivePools(), incrementBytes);
+  return arbitrator_->growCapacity(pool, getAlivePools(), incrementBytes);
 }
 
 uint64_t MemoryManager::shrinkPools(uint64_t targetBytes) {
-  return arbitrator_->shrinkMemory(getAlivePools(), targetBytes);
+  return arbitrator_->shrinkCapacity(getAlivePools(), targetBytes);
 }
 
 void MemoryManager::dropPool(MemoryPool* pool) {
   VELOX_CHECK_NOT_NULL(pool);
+  if (pool == defaultRoot_.get()) {
+    return;
+  }
   folly::SharedMutex::WriteHolder guard{mutex_};
   auto it = pools_.find(pool->name());
   if (it == pools_.end()) {
     VELOX_FAIL("The dropped memory pool {} not found", pool->name());
   }
   pools_.erase(it);
-  arbitrator_->releaseMemory(pool);
+  VELOX_CHECK_EQ(pool->currentBytes(), 0);
+  arbitrator_->shrinkCapacity(pool, 0);
+  const uint64_t capacity = pool->capacity();
+  VELOX_CHECK(
+      capacity == 0 || capacity == kMaxMemory,
```

Comment:
Curiously, when would `capacity == kMaxMemory` be valid here?

Reply:
Updated.